### PR TITLE
Cache implementations

### PR DESCRIPTION
**Pull request by PaulFarver**
- Implement `/project/:project/language/:language`
- Implement redis and filesystem cache types
- Configure cache in serve command

I've taken a different approach to dependency injection in the endpoint. Turns out echo has a "solution" to it, which I've followed: https://echo.labstack.com/guide/context

### CODE REVIEW

**Review by naueramant — APPROVED** [review]

File: internal/cache/filesystem.go [hunk 1]
```diff
@@ -1,53 +1,92 @@
...
+			return nil, ErrCacheMiss
+		}
+
+		return nil, errors.Wrap(err, "Failed to get cached file state from OS")
 	}
+
 	if time.Since(s.ModTime()) > f.ttl {
+		if err := os.Remove(filename); err != nil {
+			return nil, errors.Wrap(err, "Failed to remove expired cache file")
+		}
```
Comment (PaulFarver) [inline internal/cache/filesystem.go:57]: Why remove the expired file, when you're just going to overwrite it with the new contents from poedit immediately after?

File: internal/cache/filesystem.go [hunk 2]
```diff
@@ -1,53 +1,92 @@
...
+func (f *FilesystemCache) GetTranslation(ctx context.Context, projectID int, languageCode, format string) ([]byte, error) {
+	filename := f.key(projectID, languageCode, format)
+
+	s, err := os.Stat(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, ErrCacheMiss
+		}
+
+		return nil, errors.Wrap(err, "Failed to get cached file state from OS")
```
Comment (PaulFarver) [inline internal/cache/filesystem.go:51]:
```suggestion
	if os.IsNotExist(err) {
		return nil, ErrCacheMiss
	} else if err != nil {
		return nil, errors.Wrap(err, "Failed to get cached file state from OS")
```

File: internal/cache/filesystem.go (hunk 1 above)
Comment (PaulFarver) [reply internal/cache/filesystem.go:55]: Eh... I get it. The other thing is a bit hacky

File: internal/cache/filesystem.go (hunk 2 above)
Comment (naueramant) [reply internal/cache/filesystem.go]: Seems weird to me asking "Is THIS of type x" and then ask "does THIS even exists"

File: internal/cache/filesystem.go (hunk 1 above)
Comment (naueramant) [reply internal/cache/filesystem.go]: I think you are right. I wll remove it.

File: internal/cache/filesystem.go (hunk 2 above)
Comment (PaulFarver) [reply internal/cache/filesystem.go]: I don't think of it like that. You've already assigned `err` a value. nil just happens to be `no error`. I think the nested ifs make it messier to read.

Comment (naueramant) [reply internal/cache/filesystem.go]: Okay, in that case maybe we should just remove "else if" and just have "if" ? this should also cleanup the control flow a tiny bit.

**Review by naueramant — APPROVED** [review]

File: internal/cache/filesystem.go (hunk 2 above)
Comment (PaulFarver) [reply internal/cache/filesystem.go]: @naueramant

**Pull request merged**